run.py

```diff
@@ -883,6 +888,18 @@ def main():
     print("FAILED:  %d" % (failed))
     print("SKIPPED: %d" % (skipped))
 
+    if args.debugfailed:
+        if len(failedLogs) > 0:
+            print("")
+            print("Failed tests debug output:")
+        for dirpath in failedLogs:
+            print("- Test %s:" % os.path.basename(dirpath))
+            for r, d, f in os.walk(dirpath+"/output"):
+                for fname in f:
+                    print("  - %s" % fname)
+                    fcontents = open(dirpath+"/output/"+fname, "r")
```

Comment:
This file pointer would have to be closed. Maybe a context manager with a [`with` statement](https://www.python.org/dev/peps/pep-0343/) would be better.
```
with open(filename) as fp:
    # do something with file
# unindent shall close the file now
```

Reply:
nit: spaces around operators

Reply:
ok for both